Sources/Signal.swift

```diff
@@ -2422,6 +2422,21 @@ extension Signal where Error == NoError {
 		}
 	}
 
+	/// Promote a signal that does not generate failures into one that can.
+	///
+	/// - note: This does not actually cause failures to be generated for the
+	///         given signal, but makes it easier to combine with other signals
+	///         that may fail; for example, with operators like
+	///         `combineLatestWith`, `zipWith`, `flatten`, etc.
+	///
+	/// - parameters:
+	///   - _ An `ErrorType`.
+	///
+	/// - returns: A signal that has an instantiatable `ErrorType`.
+	public func promoteError(_: Error.Type = Error.self) -> Signal<Value, Error> {
```

Comment:
It isn't necessary, just that it allows one (brainlessly) adds `promoteError` to everything, while it doesn't create extra `Signal`s in cases with the identical error types.